### PR TITLE
woc

### PR DESCRIPTION
**Pull request by Serein207**

### CODE REVIEW

File: src/ViewModel/NextUpViewModel.cpp [hunk 1]
```diff
@@ -190,15 +190,15 @@ Song NextUpViewModel::getNextSong() {
         break;
     }
     case PlayMode::ListRepeat: {
-        // TODO
+        song = model.first();
```
Comment (ArcticLampyrid) [inline src/ViewModel/NextUpViewModel.cpp:193]: Really working?

File: src/Utility/Tools.cpp [hunk 2]
```diff
@@ -1,8 +1,35 @@
...
+    return randomGenerator.bounded(low, high);
+}
+
+QList<SongLyric> Tools::parseLyric(const QString& lyric) {
+    QList<SongLyric> lyrics;
+    QStringList lines = lyric.split('\n');
+    for (const auto& line : lines) {
+        if (line.isEmpty())
+            continue;
+        auto index = line.indexOf(']') + 1;
```
Comment (ArcticLampyrid) [inline src/Utility/Tools.cpp:29]: What about metadata?

File: src/ViewModel/NextUpViewModel.cpp (hunk 1 above)
Comment (Serein207) [reply src/ViewModel/NextUpViewModel.cpp:193]: 模拟一个循环队列，从头部pop，从尾部push

File: src/Utility/Tools.cpp (hunk 2 above)
Comment (Serein207) [reply src/Utility/Tools.cpp:29]: 决定不处理了（悲

**Pull request closed without merging**